app/src/main/java/com/sdp13epfl2021/projmag/MainActivity.kt

```diff
@@ -36,12 +39,36 @@ class MainActivity : AppCompatActivity() {
         finish()
     }
 
-    fun goToProjectsList() {
+    fun goToList(addExtras: (Intent) -> Unit) {
         val intent = Intent(this, ProjectsListActivity::class.java)
+        addExtras(intent)
         startActivity(intent)
         finish()
     }
 
+    fun handleLink() {
+        FirebaseDynamicLinks.getInstance().getDynamicLink(intent).addOnSuccessListener {
+                pendingDynamicLinkData ->
+            var dynamicLink: Uri? = null
+            if (pendingDynamicLinkData != null) {
+                dynamicLink = pendingDynamicLinkData.link!!
+            }
+            var fromLink = false
+            var projectId: String? = ""
+            if (dynamicLink != null) {
+                projectId = dynamicLink.path?.substring(11)
+                fromLink = true
+            }
+
+            goToList { i ->
+                i.putExtra("fromLink", fromLink)
+                i.putExtra("projectId", projectId)
```

Comment:
You shouldn't use hard-coded string as key. Put a public `const val ` in the `companion object` so that you can access it later from another activity.

Reply:
done